scrapyd_k8s/launcher/docker.py

```diff
@@ -21,7 +21,11 @@ def __init__(self, config):
         self._docker = docker.from_env()
 
     def get_node_name(self):
-        return os.getenv('HOSTNAME')
+        hostname = os.getenv('HOSTNAME')
+        if hostname in [None, "", "null"]:
```

Comment:
Most often HOSTNAME exists in linux based systems but it's not part of posix family standard so there can be cases when it's null, this is the source my thoughts are based on:
https://superuser.com/questions/132489/hostname-environment-variable-on-linux#comment1214603_132500

Reply:
What brings `HOSTNAME` that `gethostname()` doesn't bring? (Perhaps the ability to override the node name from the environment, but then I'd like it across launcher backends, and not provide that ability here.) So I would just use `gethostname()`. You can already override the node name in the config file.

Reply:
Yes, true, os.getenv is just a call to "third-party" program while gethostame is part of core if I understand correctly so better just use gethostname and in case it's null then use "default"?

Reply:
Well, both are from the core Python library, but `os.getenv()` looks in the environment, and `gethostname()` does a syscall to request the system hostname. The Superuser question shows that `HOSTNAME` may not be a proper environment variable (and if it is may depend on the Linux distribution or other *nix system flavour, in any case it is not part of the POSIX spec), so that you cannot really rely on it.